scheduler/replica_scheduler.go

```diff
@@ -45,13 +45,31 @@ func NewReplicaScheduler(ds *datastore.DataStore) *ReplicaScheduler {
 	return rcScheduler
 }
 
+// isFailedReplicaUnused will return true if replica was not used before and marked as failed
+func isFailedReplicaUnused(replica *longhorn.Replica, replicas map[string]*longhorn.Replica) bool {
+	isFirstAttachment := true
```

Comment:
nit: the naming `isFirstAttachment` is confusing... The purpose of `isFirstAttachment` is to check if there is any scheduled replica.

Reply:
I will remove this at next commit.

Reply:
BTW, why need to check if there is any scheduled replica here?

Reply:
Yeah, it might not need to check this now. I will test this.
At first it just give a chance to schedule the replica if all replicas `replica.Spec.HealthyAt == ""` but now use NodeId.

Reply:
Make it more simple, just to check the replica `HealtyhAt` and `FailedAt`.